msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/nestedAppAuth/NestedAppHelper.java

```diff
@@ -210,14 +217,20 @@ protected void performATSilentForNestedApp(AccountRecord accountRecord, boolean
         Assert.assertTrue(acquireTokenSilentResult.getSucceeded());
         // NAA requests must not be serviced from cache
         Assert.assertFalse(acquireTokenSilentResult.getLocalAuthenticationResult().isServicedFromCache());
-        // cannot parse jwt for MSA
+
+        validateAppIdIfRequired(acquireTokenSilentResult, NESTED_APP_CLIENT_ID);
+    }
+
+    private void validateAppIdIfRequired(@NonNull final AcquireTokenResult acquireTokenResult, @NonNull final String expectedClientId) throws ServiceException {
+        // Cannot parse jwt for MSA, so no validation for MSA
```

Comment:
Why?

Reply:
tokens issued for MSA users are always encrypted.